diff --git a/index.d.ts b/index.d.ts
index d5b2e994b42b9e7d149c984da0947cb201d5da43..0e1c79bd3d94dd5dfe6e49ea60a977788c1ba611 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -1,6 +1,6 @@
-import "chevrotain";
+import {IRecognitionException} from "chevrotain";
 
-declare namespace kdljs {
+// declare namespace kdljs {
   /**
    * A {@link https://github.com/kdl-org/kdl/blob/main/SPEC.md#document|Document}.
    */
@@ -32,7 +32,7 @@ declare namespace kdljs {
    */
   export interface NodeTypeAnnotations {
     /** The type annotation of the Node */
-    name: string;
+    name?: string;
     /** The type annotations of the {@link https://github.com/kdl-org/kdl/blob/main/SPEC.md#argument|Arguments} */
     values: string[];
     /** The type annotations of the {@link https://github.com/kdl-org/kdl/blob/main/SPEC.md#property|Properties} */
@@ -46,7 +46,7 @@ declare namespace kdljs {
 
   export interface ParseResult {
     /** Parsing errors */
-    errors: chevrotain.IRecognitionException[];
+    errors: IRecognitionException[];
     /** KDL Document */
     output?: Document;
   }
@@ -69,25 +69,25 @@ declare namespace kdljs {
     printNullArgs?: boolean;
     printNullProps?: boolean;
   }
-}
+// }
 
 /**
  * @param {string} text - Input KDL file (or fragment)
  */
-export function parse(text: string): kdljs.ParseResult;
+export function parse(text: string): ParseResult;
 
 /**
  * @param {kdljs.Document} doc - Input KDL document
  */
-export function format(doc: kdljs.Document, options?: kdljs.FormattingOptions): string;
+export function format(doc: Document, options?: FormattingOptions): string;
 
 /**
  * @param {kdljs.Document} doc - Input KDL document
  * @param {kdljs.QueryString} queryString - Query for selecting and/or transforming results
  */
-export function query(doc: kdljs.Document, queryString: kdljs.QueryString): any;
+export function query(doc: Document, queryString: QueryString): any;
 
 /**
  * @param {kdljs.Document} doc - KDL document
  */
-export function validateDocument(doc: kdljs.Document): boolean;
+export function validateDocument(doc: Document): boolean;
